trainer.py

```diff
@@ -51,7 +51,7 @@ def train_one_epoch(self) -> tuple[float, float]:
         total_loss = 0
         avg_loss = 0
         accuracy = 0
-        nof_samples = 0
+        nof_samples = len(self.train_dataset)
```

Comment:
change back to 0, and update during batch iteration
nof_samples += self.batch_size  (or something like that)